src/chem/aeronet_aod2ioda.py

```diff
@@ -0,0 +1,381 @@
+        print(latlonbox)
+        if inv_type is None:
+            inv_type = ''
+        self.url = base_url + date_portion + product + \
+            inv_type + time + latlonbox + '&if_no_html=1'
+
+    def read_aeronet(self):
+        print('Reading Aeronet Data...')
+        df = pd.read_csv(self.url,
+                         engine='python',
+                         header=None,
+                         skiprows=6,
+                         parse_dates={'time': [1, 2]},
+                         date_parser=dateparse,
+                         na_values=-999)
+        columns = self.get_columns()
+        df.columns = columns
+        df.index = df.time
+        df.rename(columns={
+            'site_latitude(degrees)': 'latitude',
+            'site_longitude(degrees)': 'longitude',
+            'site_elevation(m)': 'elevation',
+            'aeronet_site': 'siteid'
+        },
+            inplace=True)
+        df.dropna(subset=['latitude', 'longitude'], inplace=True)
+        df.dropna(axis=1, how='all', inplace=True)
+        self.df = df
+
+    def get_columns(self):
+        header = pd.read_csv(self.url, skiprows=5, header=None,
+                             nrows=1).values.flatten()
+        final = ['time']
+        for i in header:
+            if "Date(" in i or 'Time(' in i:
+                pass
+            else:
+                final.append(i.lower())
+        return final
+
+    def add_data(self,
+                 dates=None,
+                 product='AOD15',
+                 latlonbox=None,
+                 daily=False,
+                 interp_to_aod_values=None,
+                 inv_type=None,
+                 freq=None,
+                 siteid=None):
+        self.latlonbox = latlonbox
+        self.siteid = siteid
+        if dates is None:  # get the current day
+            self.dates = pd.date_range(start=pd.to_datetime('today'),
+                                       end=pd.to_datetime('now'),
+                                       freq='H')
+        else:
+            self.dates = dates
+        self.prod = product.upper()
+        if daily:
+            self.daily = 20  # daily data
+        else:
+            self.daily = 10  # all points
+        if inv_type is not None:
+            self.inv_type = 'ALM15'
+        else:
+            self.inv_type = inv_type
+        if 'AOD' in self.prod:
+            self.new_aod_values = interp_to_aod_values
+        self.build_url()
+        try:
+            self.read_aeronet()
+        except Exception:
+            print(self.url)
+        if freq is not None:
+            self.df = self.df.groupby('siteid').resample(
+                freq).mean().reset_index()
+        if self.new_aod_values is not None:
+            self.calc_new_aod_values()
+        return self.df
+
+    def calc_new_aod_values(self):
+
+        def _tspack_aod_interp(row, new_wv=[440., 470., 550., 670., 870., 1240.]):
+            try:
+                import pytspack
```

Comment:
This looks like something Barry created? Is there a more standard way to do this that doesn't require yet another thing installed?

Reply:
I am not sure how to apply tspack without installed library tspack.cpython-37m-x86_64-linux-gnu.so. So I have remove codes for AOD interpolation. Thanks.